pkg/upgrade/control_plane.go

```diff
@@ -859,6 +859,22 @@ func generateReplacementMachineName(base, upgradeID string) string {
 	return machineName + machineSuffix
 }
 
+// filterOutMachineOwners filters out owner references of kind "Machine",
+// leaving owner references of other kinds. This allows the Machine controller
+// to adopt the objects after they're created, but allows us to preserve owner
+// refs of various other kinds
+func filterOutMachineOwners(ownerRefs []metav1.OwnerReference) []metav1.OwnerReference {
+	filtered := make([]metav1.OwnerReference, 0, len(ownerRefs))
+	for _, owner := range ownerRefs {
+		if owner.Kind == "Machine" {
```

Comment:
@vincepri do you want to check api group as well?

Reply:
We could, for extra safety for sure

Reply:
Good call. Gone ahead and added that.